controllers/workspace/devworkspace_controller.go

```diff
@@ -413,6 +413,9 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		allPodAdditions = append(allPodAdditions, *routingPodAdditions)
 	}
 
+	annotationAdditions := controllerv1alpha1.PodAdditions{Annotations: workspace.Config.Workspace.PodAnnotations}
```

Comment:
I was on the fence about whether adding the `workspace.Config.Workspace.PodAnnotations` should be done here or somewhere closer to the devworkspace's [`getSpecDeployment()`](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/workspace/deployment.go#L131) function, given that we have a function [`getAdditionalAnnotations()`](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/workspace/deployment.go#L373-L389) which takes in a `DevWorkspaceWithConfig`.

However, `getAdditionalAnnotations()` is actually more like `getAdditionalDeploymentAnnotations()` since its return value is used for the [deployment annotations](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/workspace/deployment.go#L181). You would have to rename `getAdditionalAnnotations()` -> `getAdditionalDeploymentAnnotations()`, and add append the DWOC annotations to podAdditions before [`mergePodAdditions()`](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/workspace/deployment.go#L141) is called, the latter of which makes `getSpecDeployment()` messier.

Moreso, the area in the devworkspace controller where you're appending the annotation additions makes sense: `// Step four: Collect all workspace deployment contributions `

So I think this makes sense :+1: Just wanted to think this over outloud.

However, it _might_ be worth adding an extra commit to rename `getAdditionalAnnotations()` -> `getAdditionalDeploymentAnnotations()`. Though this is entirely optional.

Reply:
IMO it's ok as it is since all pod additions are collected before SyncDeploymentToCluster() in the main reconcile function https://github.com/devfile/devworkspace-operator/blob/69c21bd4ee855c9bb37db21e0eb32fcfbe36add3/controllers/workspace/devworkspace_controller.go#L471

I've added a new commit for the rename

Reply:
Agreed +1